packages/api/src/controllers/stripe.ts

```diff
@@ -0,0 +1,720 @@
+          let quantity = Math.round(usageRes.sourceSegmentsDuration / 60);
+          await req.stripe.invoiceItems.create({
+            customer: user.stripeCustomerId,
+            invoice: invoice.id,
+            currency: "usd",
+            period: {
+              start: invoice.period_start,
+              end: invoice.period_end,
+            },
+            subscription: user.stripeCustomerSubscriptionId,
+            unit_amount_decimal: product.price * 100,
+            quantity,
+            description: product.description,
+          });
+        }
+      })
+    );
+  } else if (event.type === "invoice.payment_failed") {
+    // Notify via email
+    const invoice = event.data.object;
+    const [users] = await db.user.find(
+      { stripeCustomerId: invoice.customer },
+      { useReplica: false }
+    );
+
+    // notify help@livepeer.org
+    await sendgridEmailPaymentFailed({
+      email: "help@livepeer.org",
+      sendgridApiKey: req.config.sendgridApiKey,
+      userId: users[0].id,
+      invoiceId: invoice.id,
+    });
+  }
+
+  // Return a response to acknowledge receipt of the event
+  return res.sendStatus(200);
+});
+
+async function sleep(millis) {
+  return new Promise((resolve) => setTimeout(resolve, millis));
+}
+
+// Migrate existing users to stripe
+app.post("/migrate-users-to-stripe", async (req, res) => {
+  if (req.config.stripeSecretKey != req.body.stripeSecretKey) {
+    res.status(403);
+    return res.json({ errors: ["unauthorized"] });
+  }
+
+  const [users] = await db.user.find(
+    [sql`users.data->>'stripeCustomerId' IS NULL`],
+    {
+      limit: 9999999999,
+      useReplica: false,
+    }
+  );
+
+  for (let index = 0; index < users.length; index++) {
+    let user = users[index];
+
+    const { data } = await req.stripe.customers.list({
+      email: user.email,
+    });
+
+    if (data.length === 0) {
+      // create the stripe customer
+      const customer = await req.stripe.customers.create({
+        email: user.email,
+      });
+
+      // fetch prices associated with free plan
+      const items = await req.stripe.prices.list({
+        lookup_keys: products["prod_0"].lookupKeys,
+      });
+
+      // Subscribe the user to the free plan
+      const subscription = await req.stripe.subscriptions.create({
+        cancel_at_period_end: false,
+        customer: customer.id,
+        items: items.data.map((item) => ({ price: item.id })),
+      });
+
+      // Update user's customer, product, subscription, and payment id in our db
+      await db.user.update(user.id, {
+        stripeCustomerId: customer.id,
+        stripeProductId: "prod_0",
+        stripeCustomerSubscriptionId: subscription.id,
+        stripeCustomerPaymentMethodId: null,
+      });
+
+      // sleep for a 200 ms to get around stripe rate limits
+      await sleep(200);
+    }
+  }
+
+  res.json(users);
+});
+
+app.post("/migrate-personal-users", async (req, res) => {
```

Comment:
This should be admin-only (`authorizer({anyAdmin: true})` or sth like that)

Edit: maybe its fine cause theres a requried secret below